pkg/chains/formats/slsa/v1/intotoite6.go

```diff
@@ -57,9 +59,32 @@ func (i *InTotoIte6) Wrap() bool {
 
 func (i *InTotoIte6) CreatePayload(ctx context.Context, obj interface{}) (interface{}, error) {
 	switch v := obj.(type) {
-	case *objects.TaskRunObject:
+	case *objects.TaskRunObjectV1:
+		tro := obj.(*objects.TaskRunObjectV1)
+		trV1Beta1 := &v1beta1.TaskRun{} //nolint:staticcheck
+		if err := trV1Beta1.ConvertFrom(ctx, tro.GetObject().(*v1.TaskRun)); err != nil {
+			return nil, fmt.Errorf("error converting Tekton TaskRun from version v1 to v1beta1: %s", err)
+		}
+		return taskrun.GenerateAttestation(ctx, objects.NewTaskRunObjectV1Beta1(trV1Beta1), i.slsaConfig)
+	case *objects.PipelineRunObjectV1:
+		pro := obj.(*objects.PipelineRunObjectV1)
+		prV1Beta1 := &v1beta1.PipelineRun{} //nolint:staticcheck
+		if err := prV1Beta1.ConvertFrom(ctx, pro.GetObject().(*v1.PipelineRun)); err != nil {
+			return nil, fmt.Errorf("error converting Tekton PipelineRun from version v1 to v1beta1: %s", err)
+		}
+		proV1Beta1 := objects.NewPipelineRunObjectV1Beta1(prV1Beta1)
+		trs := pro.GetTaskRuns()
+		for _, tr := range trs {
+			trV1Beta1 := &v1beta1.TaskRun{} //nolint:staticcheck
+			if err := trV1Beta1.ConvertFrom(ctx, tr); err != nil {
+				return nil, fmt.Errorf("error converting Tekton TaskRun from version v1 to v1beta1: %s", err)
+			}
+			proV1Beta1.AppendTaskRun(trV1Beta1)
+		}
```

Comment:
Can we convert the pipelinerun object instead like we're doing for the task run?

Reply:
In the code here, the pipelinerun object is converted prior (above this code block). Initially in implementing this I thought that (as you suggest) only converting the PipelineRun and creating the PipelineRunObject would be sufficient but if you do this the pipelinerun is created with no taskrun objects (which is important elsewhere in code as provenance info can come from pipelinerun sub-taskruns).  This is because the `NewPipelineRunObject[V1Beta1]` method and associated PipelineObject interface is designed to take in a PipelineRun and then append TaskRuns currently:

https://github.com/tektoncd/chains/blob/main/pkg/chains/objects/objects.go#L205-L209
https://github.com/tektoncd/chains/blob/main/pkg/reconciler/pipelinerun/pipelinerun.go#L119


The logic here follows current constructor + interface examples from the above.  I'm happy to tweak any interfaces, etc. to make this less verbose but currently IIUC there isn't a simple way to make a vX -> vY copy of PipelineRunObject without adding back the TaskRuns manually through the AppendTaskRun method